tests/test_aside.py

```diff
@@ -131,31 +136,53 @@ def test_responses(self):
         """
         Test that the responses API will show recorded events during the open period
         """
+        # The testing modulestore expects deprecated keys for some reason
+        course_id = self.aside_instance.course_key.replace(deprecated=True)
         problem_id = self.aside_instance.wrapped_block_usage_key
-        course_id = self.aside_instance.course_key
-
-        submissions = []
-        for n in range(5):
-            username = 'user_{}'.format(n)
-            email = 'user{}@email.com'.format(n)
-            answer_id = 'answer_{}'.format(n)
-            answer_text = 'Answer #{}'.format(n)
-
-            user = User.objects.create(
-                username=username,
-                email=email,
-            )
+        # replace(deprecated=True) doesn't work for BlockUsageLocator
+        problem_id = BlockUsageLocator(course_id, problem_id.block_type, problem_id.block_id, deprecated=True)
+
+        # This problem is imported into the modulestore in the setup method.
+        # It needs to be present there to allow the view function to look up problem data.
+        request = RequestFactory().request()
+        request.user = self.instructor
+        request.session = request.environ
+        problem = self.get_problem_by_id(problem_id)
+        aside_block = get_aside_from_xblock(problem, self.aside_usage_key.aside_type)
+
+        answer_id_counts = zip(range(3), range(2, 5))
+        answer_texts = [
+            'an incorrect answer',
+            'the correct answer',
+            'a different incorrect answer',
+        ]
```

Comment:
We could also do this, but it might be too fancy...

```python
answer_id_answer_counts = zip(
  range(3), 
  range(2, 5),
  ['an incorrect answer', ...]
)
```

You can take or leave this one